### PR TITLE
Yield farming user deposit UI fix

### PR DESCRIPTION
**Pull request by abhishek-01k**
Things to review:

- [x] User Deposit in Yield Farming gets null when the dapp reloads.
- [x] The dropdown on the header was not working, now its fixed. Check for mobile version too.
- [x] When you set chain to something else other than sepolia or ethereum, you get a modal in Yield Farming asking you to change the network, check if this is working fine or not. Also, check the dark mode for this modal.
- [x] When this modal appears, scroll the page and see if this overlaps the header or not.

### CODE REVIEW

**Review by mishramonalisha76 — CHANGES_REQUESTED** [review]
Switching networks pops up the metamask to change the netowrk automatically, clicking on the switch network button dosnot do anything. Is it the rigth functionality?

**Review by mishramonalisha76 — APPROVED** [review]
LGTM, but the merge conflict needs to be fixed

File: src/App.tsx [hunk 1]
```diff
@@ -81,15 +81,15 @@ const extendConsole = () => {
         window.console = {};
       }
       if (window.console[level] === 'undefined' || !window.console[level] || window.console[level] === null) {
-        window.console[level] = function () {};
+        window.console[level] = function () { };
```
Comment (rohitmalhotra1420) [inline src/App.tsx:84]: Why do we need this in the first place. Can't these be removed?

File: src/App.tsx [hunk 2]
```diff
@@ -81,15 +81,15 @@ const extendConsole = () => {
...
       }
       if (enabled) {
         if (disabledConsoles[level]) {
           window.console[level] = disabledConsoles[level];
         }
       } else {
         disabledConsoles[level] = window.console[level];
-        window.console[level] = function () {};
+        window.console[level] = function () { };
```
Comment (rohitmalhotra1420) [inline src/App.tsx:92]: Same

File: src/components/YieldFarmChainError.tsx [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
-      console.debug('Current Chain ID ', currentChainId);
-      console.debug('Chain Id to pass', chainIdToPass);
+      console.info('Current Chain ID ', currentChainId);
+      console.info('Chain Id to pass', chainIdToPass);
       switchChain(appConfig.coreContractChain);
     }
   };
 
+  useEffect(() => {
+    if (currentChainId === 1 || currentChainId === 11155111) {
```
Comment (rohitmalhotra1420) [inline src/components/YieldFarmChainError.tsx:25]: Why we are these specific values like 1 and 11155111? 

What does these justify here? 

Are we completely sure these will remain fixed? If yes then please store these values in a const with proper comments letting the developer know more about it.



```
// contants.ts

// The meaning and purpose of these id are...

const ChainIdTypeA: ChainIdType = 1;

const chainIdTypeB: ChainIdType = 11155111;

// types.ts

export type ChainIdType = 1 | 11155111

```
P.S. You could use more meaningful names.

File: src/primaries/Profile.tsx [hunk 4]
```diff
@@ -88,11 +88,11 @@ const Profile = ({ isDarkMode }) => {
     setShowDropdown(false);
   });
 
-  const ConnectWallet = ()=>{
+  const ConnectWallet = () => {
```
Comment (rohitmalhotra1420) [inline src/primaries/Profile.tsx:91]:
```suggestion
  const handleConnectWallet = () => {
```

File: src/primaries/Profile.tsx [hunk 5]
```diff
@@ -168,7 +168,7 @@ const Profile = ({ isDarkMode }) => {
           bg="linear-gradient(87.17deg, #B6A0F5 0%, #F46EF7 57.29%, #FF95D5 100%)"
           color='#FFF'
           isDarkMode={isDarkMode}
-          onClick={()=>ConnectWallet()}
+          onClick={() => ConnectWallet()}
```
Comment (rohitmalhotra1420) [inline src/primaries/Profile.tsx:171]:
```suggestion
          onClick={handleConnectWallet}
```

File: src/sections/yield/NewYieldFarming.tsx [hunk 6]
```diff
@@ -54,17 +59,22 @@ const NewYieldFarming = ({ setActiveTab }) => {
...
 
     setPUSHPoolStats({ ...pushPoolStats });
     setUserDataPush({ ...userDataPush });
-  }, [staking, pushToken, pushCoreV2, yieldFarmingLP, uniswapV2Router02Instance]);
+  }, [staking, pushToken, pushCoreV2, yieldFarmingLP, uniswapV2Router02Instance, provider]);
 
   //initiate the YieldFarmV2 data store here
   React.useEffect(() => {
+
+    if (chainId !== 1 && chainId !== 11155111) {
```
Comment (rohitmalhotra1420) [inline src/sections/yield/NewYieldFarming.tsx:74]: Same as above

File: src/singletons/YieldFarmingDataStoreV2.js [hunk 7]
```diff
@@ -321,19 +320,31 @@ export default class YieldFarmingDataStoreV2 {
...
+          currentEpochNumber = 0,//BN
+          userStaked = bn(0),//BN
+          potentialReward = bn(0),//BN
+          availableRewards = bn(0),//BN
+        } = {};
+
+        ({ epochRewards, currentEpochNumber } = await getUserPushStakingInfo(provider, this.state.account, addresses.pushCoreV2));
+
+        //these values changes if the account changes.
+        if (this.state.account !== '0x0000000000000000000000000000000000000001') {
```
Comment (rohitmalhotra1420) [inline src/singletons/YieldFarmingDataStoreV2.js:336]: Same as above. Why this exact number?

If we need it then use the above mentioned approach.

**Review by rohitmalhotra1420 — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (abhishek-01k) [reply src/App.tsx:84]: I am not sure why this is present here.

File: src/singletons/YieldFarmingDataStoreV2.js (hunk 7 above)
Comment (abhishek-01k) [reply src/singletons/YieldFarmingDataStoreV2.js]: Fixed in new commit

File: src/sections/yield/NewYieldFarming.tsx (hunk 6 above)
Comment (abhishek-01k) [reply src/sections/yield/NewYieldFarming.tsx]: Fixed in new commit

File: src/primaries/Profile.tsx (hunk 5 above)
Comment (abhishek-01k) [reply src/primaries/Profile.tsx]: Fixed in new commit

File: src/primaries/Profile.tsx (hunk 4 above)
Comment (abhishek-01k) [reply src/primaries/Profile.tsx]: Fixed in new commit

File: src/components/YieldFarmChainError.tsx (hunk 3 above)
Comment (abhishek-01k) [reply src/components/YieldFarmChainError.tsx]: Fixed in new commit

File: src/App.tsx (hunk 2 above)
Comment (abhishek-01k) [reply src/App.tsx:93]: Fixed in new commit

**Review by rohitmalhotra1420 — APPROVED** [review]

**Pull request merged**